### ISSUE TITLE
Splitting frontend and backend

### ISSUE THREAD

**Issue by Alindah** [Alindah/flask-web-dev#1]
Would I have to divide the repo I have now, which contains both the frontend and backend, into two separate repositories if I want to host each of them on a different server?

**Comment by Alindah** [comment]
Also I know I can use Netlify/Heroku to host, but I'm trying to figure out how to deploy using GCP.

I think I have to create two VMs (servers) using Google's Compute Engine, one to host the frontend and the other for backend. Then I'd have to package everything up with Docker and deploy using Kubernetes (not sure how to tie them all in yet).

Am I heading in the right direction?

Edit: Actually Compute Engine VMs are too resource intensive for what we're doing and I think Kubernetes/Docker makes them redundant. I need to use GKE to host a cluster, which will exist in the cloud. The cluster will contain at least two nodes (which are also technically VMs/servers?), one for frontend and the other for backend.

**Comment by Alindah** [comment]
Also GCP has some other web hosting APIs I can use, like Firebase, Wordpress, and Apache. Not sure if I should use one of those?

Another idea, could I just have the frontend code on github and reference them in my backend code?

Edit: Scratch this, I don't think I have to go down this path if I'm going to use Kubernetes.

**Comment by Alindah** [comment]
Ok I think I know what I have to do.

So the frontend and backend do need their own repositories. We use Kubernetes to deploy both of these servers.

The frontend server is what the client accesses. When the user submits a form, the frontend sends a POST request to the ip address of the backend server. The backend then does whatever it needs with the data and returns the result to the frontend to display to the user.

**Comment by Alindah** [comment]
I created two new repos:

- https://github.com/Alindah/fwd-frontend
- https://github.com/Alindah/fwd-backend

I ran both of them on localhost with different ports and was able to have the backend get a string from the frontend form and return the correct string length; however, it redirects to the backend server page. Not sure how to prevent that. Ideally, I want to use javascript to handle the response, and display it on the same page.

Once I've figured this all out, I'd need to go on GCP and use the GKE API to create a cluster in the cloud. The cluster will have two nodes, each with a different ip address. The ip addresses will be made static, so it doesn't change every time the nodes stop/start. We can containerize and host the backend on one of these nodes and grab its static ip and have our frontend refer to it. Then we containerize and host the frontend on another node.
